zha/application/platforms/button/__init__.py

```diff
@@ -132,6 +132,7 @@ def create_platform_entity(
 
     _attr_device_class = ButtonDeviceClass.IDENTIFY
     _attr_entity_category = EntityCategory.DIAGNOSTIC
+    _attr_translation_key = "identify"
```

Comment:
I don't think this should have a translation key for HA. It already has the "identify" device class which is enough for the name. Same as battery entities (only use device class, not another explicit "Battery" name/translation).

Reply:
You're right, this can be removed without issue.

I think originally I was setting `_attr_name = None`, which again defaulted to the device name. Fixed by not setting the attribute unless it has a value.